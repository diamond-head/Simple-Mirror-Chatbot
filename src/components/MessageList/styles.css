.message-list__container {
  padding: 16px;
  display: flex;
  flex-direction: column;
  row-gap: 16px;
  max-height: calc(100vh - 135px);
  overflow: auto;
  scroll-behavior: smooth;
}

.message-item__wrapper {
  display: flex;
}