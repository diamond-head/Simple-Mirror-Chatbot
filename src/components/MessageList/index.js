import React from 'react';
import SingleMessageBox from '../SingleMessageBox';

import "./styles.css";

const sideStyle = Object.freeze({
  left: 'flex-start',
  right: 'flex-end'
});

export default function MessageList({
  messages,
  source,
  target
}) {
  const listRef = React.useRef(null);
  const getSide = (i) => i.source === source ? "right" : "left";

  React.useEffect(() => {
    if (listRef && listRef.current && listRef.current.scrollTo) {
      listRef.current?.scrollTo(0, listRef.current.scrollHeight);
    }
  }, [messages.length]);

  return (
    <div ref={listRef} className="message-list__container">
      {messages.map((i, idx) => (
        <div key={i.source + ":" + i.target + ":" + i.timestamp}  className="message-item__wrapper" style={{ alignSelf: sideStyle[getSide(i)] }}>
          <SingleMessageBox
            content={i.message}
            side={getSide(i)}
          />
        </div>
      ))}
    </div>
  );
}
