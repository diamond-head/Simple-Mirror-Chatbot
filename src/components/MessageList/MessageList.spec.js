import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MessageList from './index';

describe("<MessageList /> component", () => {
  it("MessageList snapshot renders", () => {
    const component = renderer.create(
      <MessageList messages={[]} source="ME" target="BOT" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
