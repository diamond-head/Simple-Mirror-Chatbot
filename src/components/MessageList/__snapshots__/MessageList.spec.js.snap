// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageList /> component MessageList snapshot renders 1`] = `
<div
  className="message-list__container"
/>
`;
